runtime/src/comm/ugni/comm-ugni.c

```diff
@@ -919,19 +919,23 @@ mpool_idx_base_t rf_done_next_pool_i(void)
 // fork_reqs_map array is indexed by locale; each element [i] holds
 // locale i's fork_reqs base address.
 //
-// There is an obvious tension on FORK_T_MAX_ARG_SIZE here.  The more
-// data we can store in the fixed-sized elements of fork_reqs, the
-// fewer fork requests there will be for which the remote end has to
-// do a remote GET of an argument larger than that.  But there can be
-// many fork_reqs elements.  For a hypothetical application running on
-// 1k locales with 16 communication domains per locale and 2 fork
-// request buffers per comm domain, the current FORK_T_MAX_ARG_SIZE
-// means that each locale has about 2 MiB of space dedicated to fork
-// requests.  That's not much, but one can see that things could get
-// out of hand quickly if FORK_T_MAX_ARG_SIZE were very large.  As of
-// this writing (11/21/11) we don't have a good handle on the
-// distribution of remote fork argument sizes.  Some better
-// information about that would be nice to have.
+// There is an obvious tension on MAX_SMALL_CALL_PAYLOAD here.  The
+// more data we can store in the fixed-sized elements of fork_reqs,
+// the fewer fork requests there will be for which the remote end has
+// to do a remote GET of an argument larger than that.  But there can
+// be many fork_reqs elements.  For an application running on 1,024
+// locales with 64 communication domains per locale and 2 fork request
+// buffers per comm domain, the current MAX_SMALL_CALL_PAYLOAD means
+// that each locale has about 128 MiB of space dedicated to fork
+// requests.  That's not massive, but one can see that things could
+// get out of hand quickly if MAX_SMALL_CALL_PAYLOAD were very large.
+// The largest job we could run on an XC is around 8,096 nodes with
+// 128 communication domains, which would be 2 GiB per node.  That's
+// pretty large, but most people are running much smaller scales so we
+// believe the tradeoff is worthwhile.  As of this writing (11/30/18)
+// the typical size for remote forks for our multi-locale performance
+// benchmark kerels is between 512 and 1024 bytes. Bundles for steup
```

Comment:
Typos: 'kerels' for 'kernels', and 'steup' for (presumably) 'setup'.

Reply:
That's embarrassing. I must have left these in intentionally so you would focus on spelling errors instead of looking at the code in depth :)

Reply:
Well, you did say to look at the commits before looking at the giant PR description, which virtually guaranteed that I would do it the other way around!